src/metabase/public_settings.clj

```diff
@@ -926,8 +926,30 @@ See [fonts](../configuring-metabase/fonts.md).")
   :doc        false)
 
 (defsetting sql-parsing-enabled
-  (deferred-tru "SQL Parsing is disabled")
+  (deferred-tru "SQL Parsing is enabled. Please use [[query-analysis-native-disabled]] instead.")
+  :deprecated "0.50.14"
   :visibility :internal
   :export?    false
   :default    true
   :type       :boolean)
+
+(defsetting query-analysis-enabled
+  (deferred-tru "Whether or not we analyze any queries at all")
+  :visibility :internal
+  :export?    false
+  :default    true
+  :type       :boolean)
+
+(defsetting query-analysis-mbql-disabled
```

Comment:
I think this comment got lost across revisions, so at the risk of repeating myself:

I decided to flip to negative features for each type as if the overall setting is off, they alone cannot enable that type. But regardless of the overall setting, you know for sure that its disabled when it's off.